File: WWAlertView/WWAlertView/WWLabel.m
//
//  WWLabel.m
//  WWAlertView
//
//  Created by XingLei on 15/12/11.
//  Copyright © 2015年 XingLei. All rights reserved.
//

#import "WWLabel.h"
#import "ControlColor.h"
#import "WWConstants.h"

@implementation WWLabel

- (instancetype)init {
    self = [super init];
    if (!self) {
        return nil;
    }
    return self;
}


- (void)drawRect:(CGRect)rect {
    NSString *str = @"为了您的账户安全，我们将发送验证码到您的手机";
    NSMutableDictionary *ornamentAttributes = [NSMutableDictionary dictionary];
    ornamentAttributes[NSFontAttributeName] = [UIFont fontWithName:@"BodoniOrnamentsITCTT"
                                                              size:14];
    NSMutableParagraphStyle *ornamentParagraph = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
    ornamentParagraph.alignment = NSTextAlignmentCenter;
    ornamentAttributes[NSParagraphStyleAttributeName] = ornamentParagraph;
    ornamentAttributes[NSForegroundColorAttributeName] = [ControlColor colorWithHexString:@"#666666"];
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
    CGSize theStringSize = [str sizeWithFont:[UIFont systemFontOfSize:14]
                               constrainedToSize:self.frame.size
                                   lineBreakMode:self.lineBreakMode];
    
    [str drawInRect:CGRectMake((self.frame.size.width - theStringSize.width)/2, 10, theStringSize.width, 250) withAttributes:ornamentAttributes];
    
    NSString *str1 = nil;
    if (self.middleTitle) {
        str1 = self.middleTitle;
    }
    NSString *resultString = [str1 stringByReplacingCharactersInRange:NSMakeRange(3, 4) withString:@"****"];
    NSDictionary *attrs1 = @{
                            NSForegroundColorAttributeName : [ControlColor colorWithHexString:@"#333333"],
                            NSFontAttributeName : [UIFont systemFontOfSize:23]
                            };
    CGSize stringSize = [resultString sizeWithFont:[UIFont systemFontOfSize:23]
                           constrainedToSize:self.frame.size
                               lineBreakMode:self.lineBreakMode];
#pragma clang diagnostic pop
    if (IPHONE_6_PLUS) {
        [resultString drawInRect:CGRectMake((self.frame.size.width - stringSize.width)/2, theStringSize.height+15, self.frame.size.width, 100) withAttributes:attrs1];
    }
    else {
        [resultString drawInRect:CGRectMake((self.frame.size.width - stringSize.width)/2, theStringSize.height+5, self.frame.size.width, 100) withAttributes:attrs1];
    }
}


@end

File: WWAlerViewController_Swift/WWAlerViewController_Swift/WWLabel.swift
//
//  WWLabel.swift
//  WWAlerViewController_Swift
//
//  Created by XingLei on 15/12/15.
//  Copyright © 2015年 XingLei. All rights reserved.
//

import UIKit
import Foundation

let labelSize0:CGFloat = 14.0
let labelSize1:CGFloat = 23.0
let labelTop0:CGFloat = 20.0
let labelTop1:CGFloat = 45.0
let labelHeight:CGFloat = 250.0


let IPHONE_6: Bool = UIScreen.instancesRespondToSelector(Selector("currentMode")) ? CGSizeEqualToSize(CGSizeMake(750, 1334), (UIScreen.mainScreen().currentMode?.size)!) : false

class WWLabel: UILabel {
    
    var topTitle : NSString?
    var middleTitle : NSString?
    
    override init(frame: CGRect) {
        super.init(frame: frame)
    }
    
    override func drawRect(rect: CGRect) {
        let topString = "为了您的账户安全，我们将发送验证码到您的手机"
        let topStringFont = UIFont(name: "BodoniOrnamentsITCTT", size: labelSize0)
        let topStringColor = UIColor.blackColor()
        let topStringStyle = NSParagraphStyle.defaultParagraphStyle().mutableCopy() as? NSMutableParagraphStyle
        topStringStyle?.alignment =  NSTextAlignment.Center
        
        let topStringAttributes = [
            NSFontAttributeName: topStringFont!,
            NSForegroundColorAttributeName: topStringColor,
            NSParagraphStyleAttributeName: topStringStyle!
            ] as NSDictionary
        
        let topRect: CGRect = topString.boundingRectWithSize(frame.size, options: NSStringDrawingOptions.UsesFontLeading, attributes: topStringAttributes as? [String : AnyObject], context: nil);
        topString.drawInRect(CGRectMake((frame.width - topRect.width)/2, labelTop0 - topRect.height/2, topRect.width, labelHeight), withAttributes:topStringAttributes as? [String : AnyObject])
        
        guard let myMiddleTitle = middleTitle as? String
            else {
                return
        }
        
        let middleString = (myMiddleTitle as NSString).stringByReplacingCharactersInRange(NSMakeRange(3, 4), withString: "****")
        let middleStringAttributes = [
            NSForegroundColorAttributeName: UIColor.blackColor(),
            NSFontAttributeName: UIFont().fontWithSize(labelSize1)
            ] as NSDictionary
        let middleStringRect: CGRect = middleString.boundingRectWithSize(frame.size, options: NSStringDrawingOptions.UsesFontLeading, attributes: middleStringAttributes as? [String : AnyObject], context: nil);
        
        middleString.drawInRect(CGRectMake((frame.size.width - middleStringRect.size.width)/2, labelTop1 - middleStringRect.height/2, frame.width, labelHeight) , withAttributes:topStringAttributes as? [String : AnyObject])
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
